### ISSUE TITLE
Capture two or more devices at the same time?

### ISSUE THREAD

**Issue by chsasaw** [nwlunatic/ios-minicap#6]
@sorccu  Thank you for the great works!!
I have successed on one device.
Can I capture two or more devices in two browser tabs at the same time?

What I have done: 
`./run.sh`

to connect one device, and to connect the other one device:

`./build/ios_minicap --udid $device_id --port 9001 --resolution 400x600`

and it stop at:
```
EnableDALDevices
2016-12-02 11:01:46.403 ios_minicap[12142:338516] Available devices:
2016-12-02 11:01:46.404 ios_minicap[12142:338516] 88c2450dd52ffeb9bb24da05e8fc566bf7a2914b
2016-12-02 11:01:46.404 ios_minicap[12142:338516] 1fcab6c6785e3894ac64f4f684b603b27daa1228
2016-12-02 11:01:46.404 ios_minicap[12142:338516] CC25121R3T0F6VVD4
```

What should I do?

**Comment by nwlunatic** [comment]
You've done everything right. Screen capture of multiple devices at the same time is not supported in current ios-minicap version. We are working on it. But the exact solution is currently unknown and under research. Please, be patient )

**Comment by sorccu** [comment]
It seems like multiple devices don't work yet. Maybe in the future.

Note that you can have multiple devices per machine, but only one can be
used simultaneously.

Also thank @nwlunatic not me :)

**Comment by chsasaw** [comment]
@nwlunatic @sorccu 
Thanks!
Looking forward to the support.

**Comment by channingbreeze** [comment]
Will docker on mac work for this problem? @nwlunatic @sorccu

**Comment by sorccu** [comment]
No. It only runs on mac and docker only runs Linux containers.

**Comment by davieds** [comment]
You've done everything right. wait u team solution supported many devices

**Comment by tonylpt** [comment]
@nwlunatic, have you had a chance to look into this issue? If you have done some investigation on this, I'll be happy to help out if possible

**Comment by yuzhidi** [comment]
@nwlunatic Hi , I'm a new one to iOS, could you introduce a little about iOS-minicap principle and why not support multi-devices, I want has some try. thx!

**Comment by sorccu** [comment]
It doesn't work because the OS X API this uses only lets you use one at a
time.

**Comment by yuzhidi** [comment]
why minicap start will cause usb shutdown then restart?

**Comment by sorccu** [comment]
Because it has to change the USB configuration which causes a brief
disconnect. It's how it works.

**Comment by Tamil-Arya** [comment]
Hi, Is there any update on this issue #6 ?

**Comment by sorccu** [comment]
No. While I do have a separate solution built entirely from scratch, but it
will not be open sourced as per my employer's request.

**Comment by Tamil-Arya** [comment]
Can you hint us how you are going with another approch?

**Comment by sorccu** [comment]
No, I cannot. I can just say that it's possible.

**Comment by retd-jerome** [comment]
> No. While I do have a separate solution built entirely from scratch, but it
> will not be open sourced as per my employer's request.

Is it something you want to monetize ? Is it possible to contact you about it ?

**Comment by ACBNair** [comment]
@sorccu  How could I contact you for the multiple iOS solution?

**Comment by sorccu** [comment]
I can’t share it.

**Comment by yuzhidi** [comment]
> I can’t share it.

You have total implemented? right?

**Comment by sorccu** [comment]
It has been in production for a few years now. While I may have made it, it belongs to my previous employer.

**Comment by ACBNair** [comment]
Thanks @sorccu . Do you know how we can contact them?

**Comment by ACBNair** [comment]
@sorccu can you share the details of that , so that we can purchase it? I am new to this, so no idea about which app/lib you are talking about.

**Comment by tyrocoding** [comment]
@ACBNair Use XCUITEST to get screenshot, later convert it to jpeg images if required and then pass on the jpeg images to mjpeg server.

**Comment by retd-jerome** [comment]
@vinodpatil103 how many frame per seconds may we expect this way ?
I guess @sorccu implemented something comparable with ios-minicap framerate.

**Comment by ACBNair** [comment]
Yes I think the same. fps and performance.

**Comment by tyrocoding** [comment]
@retd-jerome @ACBNair Around 10 fps, This resolves the problem of running on multiple devices, performance would be degraded.

**Comment by ACBNair** [comment]
@nwlunatic  Hi, I saw a branch you worked on . How was it?

**Comment by KavithaRamkrishnan** [comment]
How to connect multiple devices in Minicap? It shows two or more emulator/devices error. Kindly help me to fix this. Note: Both the devices are android only.

**Comment by jjunghyup** [comment]
@kavithamanivel @ACBNair how about use this [link](https://github.com/jjunghyup/ios-screen-mirror)

**Comment by krishtoautomate** [comment]
two or more devices is also working as long as you use different ports in **run.sh** and also in **app.js**
